File: tasking/sources/free_agent.1.f90
! @@name:	free_agent.1
! @@type:	F-free
! @@operation:	compile
! @@expect:	success
! @@version:	omp_6.0
program openmp_task_example
  implicit none

  integer :: i

  ! Declare external procedures
  external :: do_background_io
  external :: do_heavy_work


  ! Create a task for background I/O
  !$omp task threadset(omp_pool)
  call do_background_io()
  !$omp end task

  ! Parallel loop for heavy work
  !$omp parallel do
  do i = 1, N
     call do_heavy_work()
  end do
  !$omp end parallel do

  ! Wait for tasks to complete
  !$omp taskwait

  ! Use results from previous tasks

end program openmp_task_example

File: tasking/sources/free_agent.1.c
/*
* @@name:	free_agent.1
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
extern void do_background_io();
extern void do_heavy_work();

int main()
{
  int i;
  // Create a task for background I/O
  #pragma omp task threadset(omp_pool)
    do_background_io();

  // Parallel loop for heavy work
  #pragma omp parallel for
   for (i = 0; i < N; i++)
     do_heavy_work();

  // Wait for tasks to complete
  #pragma omp taskwait

  // Use results from previous tasks

  return 0;
}
